metrics/web/lib/features/auth/presentation/widgets/auth_form.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:email_validator/email_validator.dart';
+import 'package:flutter/material.dart';
+import 'package:metrics/features/auth/presentation/state/user_metrics_store.dart';
+import 'package:metrics/features/auth/presentation/strings/login_strings.dart';
+import 'package:metrics/features/auth/presentation/widgets/auth_input_field.dart';
+import 'package:metrics/features/dashboard/presentation/pages/dashboard_page.dart';
+import 'package:states_rebuilder/states_rebuilder.dart';
+
+/// Shows an authentication form to sign in.
+class AuthForm extends StatefulWidget {
+  @override
+  _AuthFormState createState() => _AuthFormState();
+}
+
+/// The logic and internal state for the [AuthForm] widget.
+class _AuthFormState extends State<AuthForm> {
+  /// Global key that uniquely identifies the [Form] widget and allows validation of the form.
+  final _formKey = GlobalKey<FormState>();
+
+  /// Controls the email text being edited.
+  final TextEditingController _emailController = TextEditingController();
+
+  /// Controls the password text being edited.
+  final TextEditingController _passwordController = TextEditingController();
+
+  @override
+  Widget build(BuildContext context) {
+    return Form(
+      key: _formKey,
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: <Widget>[
+          AuthInputField(
+            key: Key(LoginStrings.email),
+            label: LoginStrings.email,
+            controller: _emailController,
+            keyboardType: TextInputType.emailAddress,
+            autofocus: true,
+            onFieldSubmitted: (String value) async => _submit(),
```

Comment:
Same suggestion with the `async` functions. And I think we are trying to use the `(_)` instead of `(String value)` to underline that we are not using this value.